esphome/components/opentherm/opentherm.h

```diff
@@ -0,0 +1,290 @@
+#endif
+#ifdef USE_SWITCH
+  void set_ch_enabled_switch(opentherm::CustomSwitch *custom_switch) { ch_enabled_switch_ = custom_switch; }
+  void set_ch_2_enabled_switch(opentherm::CustomSwitch *custom_switch) { ch_2_enabled_switch_ = custom_switch; }
+  void set_dhw_enabled_switch(opentherm::CustomSwitch *custom_switch) { dhw_enabled_switch_ = custom_switch; }
+  void set_cooling_enabled_switch(opentherm::CustomSwitch *custom_switch) { cooling_enabled_switch_ = custom_switch; }
+  void set_otc_active_switch(opentherm::CustomSwitch *custom_switch) { otc_active_switch_ = custom_switch; }
+#endif
+#ifdef USE_NUMBER
+  void set_ch_setpoint_temperature_number(opentherm::CustomNumber *number) { ch_setpoint_temperature_number_ = number; }
+  void set_ch_2_setpoint_temperature_number(opentherm::CustomNumber *number) {
+    ch_2_setpoint_temperature_number_ = number;
+  }
+  void set_dhw_setpoint_temperature_number(opentherm::CustomNumber *number) {
+    dhw_setpoint_temperature_number_ = number;
+  }
+#endif
+
+ private:
+  InternalGPIOPin *read_pin_;
+  InternalGPIOPin *write_pin_;
+  ISRInternalGPIOPin isr_read_pin_;
+
+  std::queue<uint32_t> buffer_;
+  float confirmed_dhw_setpoint_ = 0;
+  bool wanted_ch_enabled_ = false;
+  bool wanted_ch_2_enabled_ = false;
+  bool wanted_dhw_enabled_ = false;
+  bool wanted_cooling_enabled_ = false;
+  bool wanted_otc_active_ = false;
+  volatile uint32_t response_ = 0;
+  volatile OpenThermResponseStatus response_status_ = OpenThermResponseStatus::NONE;
+  volatile OpenThermStatus status_ = OpenThermStatus::NOT_INITIALIZED;
+  volatile uint32_t response_timestamp_ = 0;
+  volatile uint8_t response_bit_index_ = 0;
+  uint32_t last_millis_ = 0;
+  uint32_t start_millis_ = 0;
+  uint16_t start_interval_ = 0;
+
+  uint32_t last_millis_return_water_temp_ = 0;
+  uint32_t last_millis_boiler_water_temp_ = 0;
+  uint32_t last_millis_boiler_2_water_temp_ = 0;
+  uint32_t last_millis_flow_rate_ = 0;
+  uint32_t last_millis_ch_pressure_ = 0;
+  uint32_t last_millis_rel_mod_level_ = 0;
+  uint32_t last_millis_dhw_temp_ = 0;
+  uint32_t last_millis_dhw_2_temp_ = 0;
+  uint32_t last_millis_outside_temp_ = 0;
+  uint32_t last_millis_exhaust_temp_ = 0;
+  uint32_t last_millis_ch_max_min_temp_ = 0;
+  uint32_t last_millis_dhw_max_min_temp_ = 0;
+  uint32_t last_millis_oem_diagnostic_code_ = 0;
+  uint32_t last_millis_fault_flags_ = 0;
+  uint32_t last_millis_param_flags_ = 0;
+  uint32_t last_millis_burner_starts_ = 0;
+  uint32_t last_millis_burner_ops_hours_ = 0;
+  uint32_t last_millis_ch_pump_starts_ = 0;
+  uint32_t last_millis_ch_pump_ops_hours_ = 0;
+  uint32_t last_millis_dhw_pump_valve_starts_ = 0;
+  uint32_t last_millis_dhw_pump_valve_ops_hours_ = 0;
+  uint32_t last_millis_dhw_burner_starts_ = 0;
+  uint32_t last_millis_dhw_burner_ops_hours_ = 0;
+  uint32_t last_millis_boiler_configuration_ = 0;
+
+  void update_spread_();
+
+  void request_(OpenThermMessageType type, OpenThermMessageID id, unsigned int data);
```

Comment:
Please change all `unsigned int` to `uint32_t`
```suggestion
  void request_(OpenThermMessageType type, OpenThermMessageID id, uint32_t data);
```